biz.aQute.tester.junit-platform/src/aQute/tester/junit/platform/Activator.java

```diff
@@ -295,12 +297,26 @@ public void removedBundle(Bundle bundle, BundleEvent event, Bundle object) {
 		};
 		tracker.open();
 
+		if (Boolean.valueOf(context.getProperty("launch.services"))) {
+			try {
+				GogoCommandlet cmd = new GogoCommandlet();
+				Hashtable<String, Object> properties = new Hashtable<>();
+				properties.put("osgi.command.function", new String[] {
+					"runTests"
+				});
+				properties.put("osgi.command.scope", "tester");
+
+				trace("starting gogo commandlet");
+				context.registerService(Object.class, cmd, properties);
+			} catch (NoClassDefFoundError e) {}
```

Comment:
Can you even get an NCDFE here? The only gogo type is Description which is only used as an annotation and the class loader will be happy to load the GogoCommandlet class even if the annotations are not available. So I don't think you can ever get the error.

Reply:
Well I've learned something. I didn't know that the classloader would ignore annotations if their type was missing. I've removed it and the tests pass. In fact, it didn't occur to me (or at least I never seriously investigated) that the gogo-specific tests were running and passing even though I hadn't added gogo to the command line.

And this is also a lesson to me, that this is the kind of thing that happens when you don't do proper TDD! I obviously never attempted (or at least never succeeded) to write that failing test to trigger the NCDFE.